From 2fed6038bb9c30b903464b838b05a2ceb3412a6b Mon Sep 17 00:00:00 2001
From: WANG Chao <chaowang@redhat.com>
Date: Fri, 4 Apr 2014 14:37:42 +0800
Subject: [PATCH] dracut-pre-pivot pulls in remote-fs.target

It turns out that commit f30b74e (dracut-initqueue service runs before
remote-fs-pre.target) is partial fix for remote fs mounts. Because no
one pulls in remote-fs.target, we can never start remote fs mounts.
Now pull in remote-fs.target in dracut-pre-pivot.

Signed-off-by: WANG Chao <chaowang@redhat.com>
(cherry picked from commit ce7f408dfd473b3c2f4fdb9ddc2b2131e08007d3)
---
 modules.d/98systemd/dracut-pre-pivot.service | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules.d/98systemd/dracut-pre-pivot.service b/modules.d/98systemd/dracut-pre-pivot.service
index 6db1f2c..d7c7b1d 100644
--- a/modules.d/98systemd/dracut-pre-pivot.service
+++ b/modules.d/98systemd/dracut-pre-pivot.service
@@ -13,6 +13,8 @@ Documentation=man:dracut-pre-pivot.service(8)
 After=initrd.target initrd-parse-etc.service sysroot.mount
 After=dracut-initqueue.service dracut-pre-mount.service dracut-mount.service
 Before=initrd-cleanup.service
+Wants=remote-fs.target
+After=remote-fs.target
 ConditionPathExists=/etc/initrd-release
 ConditionDirectoryNotEmpty=|/lib/dracut/hooks/pre-pivot
 ConditionDirectoryNotEmpty=|/lib/dracut/hooks/cleanup
